pkg/lib/queueinformer/queueinformer.go

```diff
@@ -105,6 +105,13 @@ func (q *QueueInformer) metricHandlers() *cache.ResourceEventHandlerFuncs {
 	}
 }
 
+func NewQueue(ctx context.Context, options ...Option) (*QueueInformer, error) {
```

Comment:
Having a different constructor is fine, but it doesn't look like this is doing anything different from `NewQueueInformer(...)`. If the goal is to have a QueueInformer that doesn't have an informer behind it, then I think what you want to do is remove the [non-nil informer/indexer test](https://github.com/operator-framework/operator-lifecycle-manager/blob/86bac67214a58e64a3000f6aae50919380c3990a/pkg/lib/queueinformer/config.go#L54) in the `queueInformerConfig.validate()` method, or create a new validate method that ignores that check. Additionally, you would need to detect a [nil indexer](https://github.com/operator-framework/operator-lifecycle-manager/blob/86bac67214a58e64a3000f6aae50919380c3990a/pkg/lib/queueinformer/queueinformer_operator.go#L251) and attempt to use the resource nested in the event, rather than getting it from the indexer.

Reply:
I ended up creating a ResourceEvent (with type deleted) and putting that on the queue as you suggested out of band.